src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitCleanProperties = sourceString
+    .split(';')
+    .map((dirtyProperty) => dirtyProperty.trim());
+
+  const validPropertyStrings = splitCleanProperties.filter((property) => {
+    return property !== '';
+  });
+
+  const properties = validPropertyStrings.map(propertyStringToObject);
```

Comment:
You can call this function `propertyStringToObject` inside `reduce` below, to avoid using useless loop

Reply:
👍